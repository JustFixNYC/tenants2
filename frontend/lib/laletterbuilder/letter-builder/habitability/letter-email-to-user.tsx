import React from "react";

import { asEmailStaticPage } from "../../../static-page/email-static-page";
import { li18n } from "../../../i18n-lingui";
import { t, Trans } from "@lingui/macro";
import { USPS_TRACKING_URL_PREFIX } from "../../../../../common-data/loc";
import { HtmlEmail } from "../../../static-page/html-email";
import { TransformSession } from "../../../util/transform-session";
import { AllSessionInfo } from "../../../queries/AllSessionInfo";

type EmailBodyProps = {
  firstName: string;
  trackingNumber?: string;
  isInLosAngeles: boolean;
};

const EmailBody: React.FC<EmailBodyProps> = (props) => (
  <>
    <Trans>
      <p>Hello {props.firstName},</p>
      <p>
        You've sent your Notice To Repair letter. Attached to this email is a
        PDF copy for your records.
      </p>
    </Trans>
    {props.trackingNumber && (
      <p>
        <Trans>
          You can also track the delivery of your letter using USPS Tracking:
        </Trans>{" "}
        <a
          data-jf-show-href-only-in-plaintext
          href={USPS_TRACKING_URL_PREFIX + props.trackingNumber}
        >
          {props.trackingNumber}
        </a>
      </p>
    )}
    {props.isInLosAngeles ? (
      <>
        <p>
          <Trans>
            You can contact Strategic Actions for a Just Economy (SAJE) - a
            501c3 non-profit organization in South Los Angeles.
          </Trans>
        </p>
        <blockquote>
          <em>
            <Trans id="norent.sajeBlockQuote">
              Since 1996 SAJE has been a force for economic justice in our
              community focusing on tenant rights, healthy housing, and
              equitable development. SAJE believes that the fate of city
              neighborhoods should be decided by those who dwell there, and
              convenes with other organizations to ensure this occurs in a
              manner that is fair, replicable, and sustainable. Housing is a
              human right.
            </Trans>
          </em>
        </blockquote>
        <p>
          <strong>
            <Trans id="norent.sajePhoneCalls">
              Strategic Actions for a Just Economy (SAJE) is available for phone
              calls at (213) 745-9961, Monday-Friday from 10:00am-6:00pm.
            </Trans>
          </strong>
        </p>
      </>
    ) : (
      <></>
    )}
  </>
);

export const HabitabilityLetterEmailToUserBody: React.FC<{}> = () => {
  return (
    <TransformSession
      transformer={(session: AllSessionInfo): EmailBodyProps | null => {
        const { firstName, onboardingInfo } = session;
        if (!(onboardingInfo && firstName)) return null;
        return {
          trackingNumber: session.habitabilityLatestLetter?.trackingNumber,
          isInLosAngeles: !!onboardingInfo.isInLosAngeles,
          firstName,
        };
      }}
    >
      {(bodyProps) => <EmailBody {...bodyProps} />}
    </TransformSession>
  );
};

export const HabitabilityLetterEmailToUserStaticPage = asEmailStaticPage(
  (props) => (
    <HtmlEmail
      subject={li18n._(
        t`Your Notice to Repair letter and important next steps`
      )}
    >
      <HabitabilityLetterEmailToUserBody />
    </HtmlEmail>
  )
);
