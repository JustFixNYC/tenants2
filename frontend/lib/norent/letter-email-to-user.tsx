import React from "react";

import { useContext } from "react";
import { AppContext } from "../app-context";
import {
  EmailSubject,
  asEmailStaticPage,
} from "../static-page/email-static-page";
import { NorentRoutes } from "./routes";
import { li18n } from "../i18n-lingui";
import { t, Trans } from "@lingui/macro";
import { USPS_TRACKING_URL_PREFIX } from "../../../common-data/loc.json";

export const NorentLetterEmailToUser: React.FC<{}> = () => {
  const { session, server } = useContext(AppContext);
  const letter = session.norentLatestLetter;
  const faqURL = `${server.originURL}${NorentRoutes.locale.faqs}`;

  return (
    <>
      <EmailSubject value={li18n._(t`Here's a copy of your NoRent letter`)} />
      <Trans>
        <p>Hello {session.firstName},</p>
        <p>
          You've sent your NoRent letter. Attached to this email is a PDF copy
          for your records.
        </p>
      </Trans>
      {letter?.trackingNumber && <p>
        <Trans>
          You can also track the delivery of your letter using USPS Tracking:
      </Trans>{" "}
        {USPS_TRACKING_URL_PREFIX + letter.trackingNumber}
      </p>}
      <p>
        <Trans>
          To learn more about what to do next, check out our FAQ page: {faqURL}
        </Trans>
      </p>
    </>
  );
};

export const NorentLetterEmailToUserStaticPage = asEmailStaticPage(
  NorentLetterEmailToUser
);
