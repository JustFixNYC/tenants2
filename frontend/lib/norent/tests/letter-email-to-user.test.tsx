import React from "react";
import { AppTesterPal } from "../../tests/app-tester-pal";
import { NorentLetterEmailToUser } from "../letter-email-to-user";

describe("NorentLetterEmailToUser", () => {
  

  it("works", () => {
    const pal = new AppTesterPal(<NorentLetterEmailToUser />, {
      session: { firstName: "Boop" },
    });
    expect(pal.rr.container).toMatchSnapshot();
  });
});
