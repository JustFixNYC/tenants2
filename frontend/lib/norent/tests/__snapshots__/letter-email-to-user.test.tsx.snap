// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NorentLetterEmailToUser works 1`] = `
<div>
  <p
    class="jf-email-subject"
  >
    Subject: 
    Here's a copy of your NoRent letter
  </p>
  <p>
    Hello Boop,
  </p>
  <p>
    You've sent your NoRent letter. Attached to this email is a PDF copy for your records.
  </p>
  <p>
    To learn more about what to do next, check out our FAQ page: https://myserver.com/en/faqs
  </p>
</div>
`;
